### PR TITLE
Add hyperapp by single package.json w/ npm auto-update

### PR DESCRIPTION
**Pull request by jbucaran**
Pull request for issue:  #10498
Related issue(s): https://github.com/cdnjs/cdnjs/pull/10608

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
* Git repository (required): https://github.com/hyperapp/hyperapp
* Official website (optional, not the repository):
* NPM package url (optional): https://www.npmjs.com/package/hyperapp
* License and its reference: https://github.com/hyperapp/hyperapp/blob/master/LICENSE
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 43
  * Count of stars: 1205
  * Count of forks: 71
* NPM download stats (optional):
  * Downloads in the last day: 262
  *  Downloads in the last week: 702
  * Downloads in the last month: 1,175

# Essential checklist
 * [x] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by jbucaran** [comment]
Hi @hare1039. Did as requested. Let me know if this is good.

**Comment by jbucaran** [comment]
Also, the name of the distributed file is `hyperapp.js`, but I think you wanted me to enter

```
"filename": "hyperapp.min.js"
```

File: ajax/libs/hyperapp/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "hyperapp",
+  "filename": "hyperapp.min.js",
+  "description": "1kb JavaScript library for building modern UI applications",
+  "version": "0.1.0",
```
Comment (hare1039) [inline ajax/libs/hyperapp/package.json]: Just remove this line `"version": "0.1.0"` than it should be all good!

**Comment by jbucaran** [comment]
@hare1039 Version field removed. Let me know if `filename` looks good and if you need anything else.

**Comment by hare1039** [comment]
@jbucaran `hyperapp.js` is minified, so I think `filename` needs to change to `hyperapp.js`.
And I think we'd better wait for senior members to check if its good.
(I'm a newbie in cdnjs)

@pvnr0082t Could you provide some help in this PR?

**Comment by jbucaran** [comment]
@hare1039 Yes, hyperapp.js is already minified, let me change it to point to the right file.

**Comment by jbucaran** [comment]
@hare1039 Done.

**Comment by hare1039** [comment]
@jbucaran Great!
Let's wait for other member's review!

File: ajax/libs/hyperapp/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  "filename": "hyperapp.js",
+  "description": "1kb JavaScript library for building modern UI applications",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/hyperapp/hyperapp.git"
+  },
+  "author": {
+    "name": "Jorge Bucaran"
+  },
```
Comment (pvnr0082t) [inline ajax/libs/hyperapp/package.json]: Let's use `"author": "Jorge Bucaran"` here.

File: ajax/libs/hyperapp/package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "template string",
+    "hyperscript",
+    "vanilla"
+  ],
+  "npmName": "hyperapp",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*"
```
Comment (pvnr0082t) [inline ajax/libs/hyperapp/package.json]: Because there are `*gz` files in old versions, let's use `**/*.!(*gz)` instead.

**Comment by jbucaran** [comment]
@hare1039 @pvnr0082t Thanks. I completed the requested changes.

**Comment by jbucaran** [comment]
@pvnr0082t HyperApp also has a distribution with a sourcemap. Does cdnjs picks that up automatically? Or do I need to update the package json?

**Comment by PeterDaveHello** [comment]
@jbucaran we'll grab it without problem, only gz files will be excluded :+1:

**Comment by jbucaran** [comment]
Hi @PeterDaveHello! Would you mind having a look and see if I made a mistake here? I've published a new version to npm, but the autoupdate feature does not seem to be pulling the new version.

Cheers! 👋

**Pull request merged**